app/controllers/api/base_controller.rb

```diff
@@ -45,7 +45,7 @@ class BaseController < ActionController::API
     rescue_from(UnsupportedMediaTypeError)      { |e| api_error(:unsupported_media_type, e) }
 
     def index
-      klass = collection_class(@req.subject)
+      klass = collection_param_class(@req.subject, params[:type])
```

Comment:
call question: will params[:type] end up in the query?

(this may cancel the return values out)